File: flutter_todo/lib/todo/todo.service.dart
import 'package:flutter_todo/todo/todo.model.dart';

import 'package:http/http.dart' as http;
import 'dart:convert' as convert;

class TodoService {
  String URL = 'http://localhost:3000/todo';

  Future<List<Todo>> listAll() async {
    http.Response response = await http.get(URL);
    if (response.statusCode == 200 || response.statusCode == 201) {
      Iterable list = convert.json.decode(response.body);
      return list.map((i) => Todo.fromJson(i)).toList();
    }
  }

  Future<bool> create(String description) async {
    http.Response response =
        await http.post(URL, body: {'description': description});
    return (response.statusCode == 200 || response.statusCode == 201);
  }

  Future<bool> remove(int id) async {
    http.Response response = await http.delete('${URL}/${id}');
    return (response.statusCode == 200 || response.statusCode == 201);
  }

  Future<bool> update(Todo _todo) async {
    http.Response response = await http.put('${URL}', body: {
      'id': _todo.id.toString(),
      'description': _todo.description,
      'done': _todo.done.toString()
    });
    return (response.statusCode == 200 || response.statusCode == 201);
  }
}

File: angular-todo(Frontend)/src/app/todo/todo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Todo } from './todo.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TodoService {

  private URL = "http://localhost:3000";

  constructor(private httpClient: HttpClient) { }

  create(description: string): Observable<Todo> {
    return this.httpClient.post<Todo>(`${this.URL}/todo`, { description });
  }

  update(todo: Todo): Observable<Todo[]> {
    return this.httpClient.put<Todo[]>(`${this.URL}/todo`, todo);
  }

  delete(id: number): Observable<any> {
    return this.httpClient.delete(`${this.URL}/todo/${id}`);
  }

  findAll(): Observable<Todo[]> {
    return this.httpClient.get<Todo[]>(`${this.URL}/todo`);
  }

}
